src/main/java/org/cascadebot/cascadebot/data/objects/GuildPermissions.java

```diff
@@ -5,9 +5,110 @@
 
 package org.cascadebot.cascadebot.data.objects;
 
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.Channel;
+import net.dv8tion.jda.core.entities.Member;
+import org.cascadebot.cascadebot.Environment;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.permissions.Security;
+import org.cascadebot.cascadebot.permissions.objects.Group;
+import org.cascadebot.cascadebot.permissions.objects.PermissionAction;
+import org.cascadebot.cascadebot.permissions.objects.PermissionMode;
+import org.cascadebot.cascadebot.permissions.objects.User;
+import org.cascadebot.shared.SecurityLevel;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
 public class GuildPermissions {
 
+    private PermissionMode mode = PermissionMode.HIERARCHICAL;
```

Comment:
Isn't the default going to be `MOST_RESTRICTIVE` as it's easier to work with?

Reply:
Could be, I'd like a discussion about this